WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -231,7 +231,13 @@ def update
   end
 
   def my_competitions
-    @competitions = (current_user.delegated_competitions + current_user.organized_competitions + current_user.competitions_registered_for).reject(&:is_over?).uniq.sort_by(&:start_date).reverse!
+    @competitions = (current_user.delegated_competitions + current_user.organized_competitions + current_user.competitions_registered_for)
+    if current_user.person
+      @competitions += current_user.person.competitions
+    end
+    @competitions = @competitions.uniq.sort_by(&:start_date).reverse!
+    @future_competitions = @competitions.reject(&:is_over?)
+    @past_competitions = @competitions.select(&:is_over?)
```

Comment:
I think there's no guarantee here that `@future_competitions` and `@past_competitions` is unique or sorted. You should probably add a `@competitions = @competitions.uniq` first.